controllers/utils/utils.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022.
```

Comment:
Both the cloud package and the controllers package need these utils, so I put them in their own package to avoid circular imports.

Now that I understand all this stuff better, I no longer believe the pkg/cloud separation is the best pattern to organize by, but here we are. I doubt I'll get to it, it's definitely not a priority, but it would be nice to rethink and reorg.